lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,331 @@
+        accountId: store.accountId,
+        child: NewDmPicker(pageContext: context))));
+}
+
+class NewDmPicker extends StatefulWidget {
+  const NewDmPicker({
+    super.key,
+    required this.pageContext
+  });
+
+  final BuildContext pageContext;
+
+  @override
+  State<NewDmPicker> createState() => _NewDmPickerState();
+}
+
+class _NewDmPickerState extends State<NewDmPicker> with PerAccountStoreAwareStateMixin<NewDmPicker> {
+  late TextEditingController searchController;
+  final Set<int> selectedUserIds = {};
+  List<User> filteredUsers = [];
+
+  @override
+  void initState() {
+    super.initState();
+    searchController = TextEditingController()..addListener(_handleSearchUpdate);
+  }
+
+  @override
+  void dispose() {
+    searchController.dispose();
+    super.dispose();
+  }
+
+  @override
+  void onNewStore() {
+    final store = PerAccountStoreWidget.of(context);
+    _updateFilteredUsers(store);
+  }
+
+  void _handleSearchUpdate() {
+    final store = PerAccountStoreWidget.of(context);
+    _updateFilteredUsers(store);
+  }
+
+  // Function to sort users based on recency of DM's
+  void _updateFilteredUsers(PerAccountStore store) {
+    setState(() {
+      filteredUsers = store.allUsers
+        .where((user) =>
+          user.userId != store.selfUserId &&
+          user.fullName.toLowerCase().contains(searchController.text.toLowerCase()))
+        .toList()
+        ..sort((a, b) {
+          final recentDms = store.recentDmConversationsView;
+          final aLatestMessageId = recentDms.latestMessagesByRecipient[a.userId];
+          final bLatestMessageId = recentDms.latestMessagesByRecipient[b.userId];
+
+          if (aLatestMessageId != null && bLatestMessageId != null) {
+            return bLatestMessageId.compareTo(aLatestMessageId);
+          }
+          if (aLatestMessageId != null) return -1;
+          if (bLatestMessageId != null) return 1;
+          return 0;
+        });
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        _NewDmHeader(
+          selectedUserIds: selectedUserIds,
+          onClose: () => Navigator.pop(context)),
+        _NewDmSearchBar(
+          searchController: searchController,
+          selectedUserIds: selectedUserIds),
+        Expanded(child: _NewDmUserList(
+          filteredUsers: filteredUsers,
+          selectedUserIds: selectedUserIds,
+          onUserTapped: (userId) {
+            setState(() {
+              if (selectedUserIds.contains(userId)) {
+                selectedUserIds.remove(userId);
+              } else {
+                selectedUserIds.add(userId);
+              }
+            });
+          })),
+      ]);
+  }
+}
+
+class _NewDmHeader extends StatelessWidget {
+  const _NewDmHeader({
+    required this.selectedUserIds,
+    required this.onClose,
+  });
+
+  final Set<int> selectedUserIds;
+  final VoidCallback onClose;
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
nit: move `build` to the end of the class.

See also: [Flutter style guide](https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#order-other-class-members-in-a-way-that-makes-sense)